database/src/main/resources/db/migration/V1_75__Add_function_for_associated_addresses.sql

```diff
@@ -0,0 +1,112 @@
+SELECT 'Add function get_tx_associated_values()' AS comment;
+
+drop function if exists get_tx_associated_values(text, integer);
+
+create or replace function get_tx_associated_values(tx_hash text, tx_height integer)
+    returns TABLE(value varchar, type varchar)
+    language plpgsql as
+$$
+BEGIN
+    RETURN QUERY
+-- BASE tx info
+        with tx AS (
+            select id from tx_cache where hash = tx_hash and height = tx_height
+        )
+-- denom match
+        select
+            tmj.denom as value,
+            'DENOM' as type
+        from tx_marker_join tmj , tx
+        where tmj.tx_hash_id = tx.id
+        union all
+-- nft uuids
+        select
+            tnj.metadata_uuid as value,
+            tnj.metadata_type as type
+        from tx_nft_join tnj , tx
+        where tnj.tx_hash_id = tx.id
+        union all
+-- nft scope addr
+        select
+            ns.address as value,
+            tnj.metadata_type as type
+        from tx_nft_join tnj
+                 join nft_scope ns ON tnj.metadata_type = 'SCOPE' AND tnj.metadata_id = ns.id, tx
+        where tnj.tx_hash_id = tx.id
```

Comment:
It's best practice to only but join criteria in the from/joins and leave other filtering criteria in the where clause. I can't remember the exact reasons why though other than it can sometimes lead to unexpected results.

```suggestion
        from tx_nft_join tnj
                 join nft_scope ns ON tnj.metadata_id = ns.id
                 join tx ON tnj.tx_hash_id = tx.id
        where tnj.metadata_type = 'SCOPE'
```